### DISCUSSION TITLE
disconnected multiple daemons

### DISCUSSION THREAD

**Discussion by mark-hahn** [Flexget/Flexget#4070] [General] [answered]
<cross-posted from reddit>

I have a problem with daemons. I was having unexpected downloads with an old daemon. I killed my daemon and downloads were still happening. Doing a ps aux revealed that an old daemon was running independently. I killed it, started a new one, and things got back to normal.

Later I tried to do a reload and it said my daemon wasn't running even though some downloads had just happened. I did a ps aux and saw that there was actually one running. I killed that deamon process and started a new daemon and everything was fixed. Does anyone know what might be going on? Each time I do any operation I have to do a ps aux and make sure one and only one is running.

flexget -V --> 3.1.135

commands I use ...
~/flexget/bin/flexget daemon start -d
~/flexget/bin/flexget daemon stop
~/flexget/bin/flexget daemon reload

**Comment by gazpachoking** [comment]
The way it determines if a daemon is running is with a lock file that is named after and sits in the same folder as your config file. If you have multiple config files in different directories, you could possibly be starting multiple daemons running against different config files. Or if you are somehow deleting the lock file, it could lose track of the daemon. Don't have many ideas other than that.

**Comment by mark-hahn** [comment]
Thanks.  I'll study my config setup.  How does it know which config folder to use?  

Unless the problem is clear I'm thinking of backing up the db and config file then reinstalling.  Does it sound like that would work?

**Comment by mark-hahn** [comment]
This is my `.config/flexget/` directory.  I don't see anything like this anywhere else. The `config-lock` file is present if and only if the daemon is running.   I don't see anything wrong, do you?   Should I try reinstalling after saving `config.yml`, `db-config-jobs.sqlite`, and `db-config.sqlite`?
```
.config-lock
config.yml
db-config-jobs.sqlite
db-config.sqlite
flexget.log
```

It just occured to me that if the `.config-lock` file was accidently deleted it would create exactly my situation.  Right?

**Reply by gazpachoking** [reply]
> It just occured to me that if the `.config-lock` file was accidently deleted it would create exactly my situation. Right?

Yep.

> This is my `.config/flexget/` directory. I don't see anything like this anywhere else. The `config-lock` file is present if and only if the daemon is running. I don't see anything wrong, do you?

I don't see anything wrong. My theory was only that you could have a `config.yml` in some other directories. If you started flexget while you were in those directories it would be used instead of your `.conifg/flexget/config.yml`. See [here](https://flexget.com/en/Configuration#location) for all the locations flexget looks for config files.

**Comment by mark-hahn** [comment]
Mucho thanks.  I've done experiments and I'm sure the lock file is being deleted.  How the hell this is happening is beyond me.  I'm considering writing a script that constantly checks the file and emails me when it disappears.

**Comment by mark-hahn** [comment, marked as answer]
For any lurkers, I finally fully understand the problem. @gazpachoking was correct from the beginning.  There were multiple directories containing config.yml.  I had a symbolic link in my home directory pointing to the real config.yml.  When I was testing this theory I used the `find` command which by ignores links by default.  So I only saw the one.
